www/src/context/SettingsContext.tsx

```diff
@@ -44,12 +44,14 @@ function SettingsContextProvider({ children }) {
     }
     setSettings(prevState => ({ ...prevState, [key]: value }));
     global.localStorage.setItem('pgn__settings', JSON.stringify({ ...settings, [key]: value }));
-    global.analytics.track(`${key[0].toUpperCase() + key.slice(1)} change`, { [key]: value });
+    global.analytics.track(`openedx.paragon.docs.settings.${key}.changed`, { [key]: value });
```

Comment:
I think we should try to send additional data in the form `{ value: 'some_value' }` to be consistent, e.g. here when language is changed you currently send `'openedx.paragon.docs.settings.language.changed', { language: 'Spanish' }`, instead we could send `'openedx.paragon.docs.settings.language.changed', { value: 'Spanish' }`, where event name describes which value changed and additional data provides new value. Same could be done for other events.

@adamstankiewicz what do you think? I actually don't know how it affects Snowflake dashboard, if it does at all 😅

Reply:
Hmm, good call out @viktorrusakov. Given that the `*.language.changed` is in the event name, I don't believe we'd ever need to do a `group by` on the `language`/`value` attribute in the event properties here.

If we had this event named something like `openedx.paragon.docs.settings.changed` and passed `{ settingsAttribute: 'language', value: 'Spanish' }`, then we might want the `value` attribute because that SQL query would definitely require a `group by` on a attribute that's consistent across all setting fields.

When looking at each explicit event per setting field (e.g., `openedx.paragon.docs.settings.language.changed`), the SQL query can just `group by` on the known  event-specific property (e.g., `language`).